### PR TITLE
Expose streams in public replace APIs

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Contributes to #925

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/replace/replace_tests.cpp [hunk 1]
```diff
@@ -47,8 +47,11 @@ TEST_F(ReplaceErrorTest, SizeMismatch)
   cudf::test::fixed_width_column_wrapper<int32_t> values_to_replace_column{{10, 11, 12, 13}};
   cudf::test::fixed_width_column_wrapper<int32_t> replacement_values_column{{15, 16, 17}};
 
-  EXPECT_THROW(cudf::find_and_replace_all(
-                 input_column, values_to_replace_column, replacement_values_column, mr()),
+  EXPECT_THROW(cudf::find_and_replace_all(input_column,
+                                          values_to_replace_column,
+                                          replacement_values_column,
+                                          cudf::test::get_default_stream(),
+                                          mr()),
```
Comment (davidwendt) [inline cpp/tests/replace/replace_tests.cpp:54]: Why was mr being passed here? Maybe this should be removed?

Comment (vyasr) [reply cpp/tests/replace/replace_tests.cpp:54]: Good point, I assumed this was a meaningful mr but it's just pulling [the mr set on the base fixture](https://github.com/rapidsai/cudf/blob/branch-23.10/cpp/include/cudf_test/base_fixture.hpp#L59). Do you know why that mr even exists? It's used pretty sparingly, and given its definition it seems like we could probably just remove it altogether. What do you think?

Comment (davidwendt) [reply cpp/tests/replace/replace_tests.cpp:54]: I think we can remove it. I'm not seeing a purpose for it here.

Comment (vyasr) [reply cpp/tests/replace/replace_tests.cpp:54]: I'll update these files in this PR and make a follow-up to remove from the fixture.

Comment (vyasr) [reply cpp/tests/replace/replace_tests.cpp]: Done.

File: cpp/src/replace/nans.cu [hunk 2]
```diff
@@ -202,7 +204,7 @@ std::unique_ptr<column> normalize_nans_and_zeros(column_view const& input,
 
   // from device. unique_ptr which gets automatically cleaned up when we leave.
   auto out_view = out->mutable_view();
-  normalize_nans_and_zeros(out_view, stream);
+  detail::normalize_nans_and_zeros(out_view, stream);
```
Comment (mythrocks) [inline cpp/src/replace/nans.cu:207]: I'm a little confused with this part. The enclosing function is in `cudf::detail`. Is `detail::` necessary here? 
The intended target is the preceding function in the same namespace, no?

Comment (bdice) [reply cpp/src/replace/nans.cu:207]: It makes that namespace lookup a bit more explicit. I think this is slightly better than before.

**Review by bdice — APPROVED** [review]
Good reviews @davidwendt and @mythrocks -- I think this is ready.

**Review by mythrocks — APPROVED** [review]

File: cpp/tests/streams/replace_test.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  auto high = cudf::numeric_scalar<int>(7, true, cudf::test::get_default_stream());
+  cudf::clamp(input, low, high, cudf::test::get_default_stream());
+}
+
+TEST_F(ReplaceTest, NormalizeNansAndZeros)
+{
+  auto nan          = std::numeric_limits<double>::quiet_NaN();
+  auto input_column = cudf::test::make_type_param_vector<double>({-0.0, 0.0, -nan, nan, nan});
+  cudf::test::fixed_width_column_wrapper<double> input(input_column.begin(), input_column.end());
+  cudf::normalize_nans_and_zeros(static_cast<cudf::column_view>(input),
```
Comment (davidwendt) [inline cpp/tests/streams/replace_test.cpp:98]:
```suggestion
  cudf::normalize_nans_and_zeros(input,
```
Just a minor nit. I don't think this cast is needed.

**Review by davidwendt — APPROVED** [review]

**Comment by vyasr** [comment]
Can't find the comment anymore, but for the record the reason the tests include explicit casts from `column_wrapper` to `cudf::column_view ` and `cudf::mutable_column_view` when calling `normalize_nans_and_zeros` is that both casts operators are defined and the function also has overloads for both, so the calls are ambiguous.

**Pull request merged**